EntityRecognitionCustomSkill/function_app.py

```diff
@@ -0,0 +1,136 @@
+from azure.core.credentials import AzureKeyCredential
+
+app = func.FunctionApp(http_auth_level=func.AuthLevel.ANONYMOUS)
+
+# A healthcheck endpoint. Important to make sure that deployments are healthy.
+# It can be accessed via <base_url>/api/health
+@app.route(route="health", auth_level=func.AuthLevel.ANONYMOUS)
+def HealthCheck(req: func.HttpRequest) -> func.HttpResponse:
+    logging.info("Calling the healthcheck endpoint")
+    response_body = {"status": "Healthy"}
+    response = func.HttpResponse(
+        json.dumps(response_body, default=lambda obj: obj.__dict__)
+    )
+    response.headers["Content-Type"] = "application/json"
+    return response
+
+
+@app.function_name(name="EntityRecognition")
+@app.route(route="entity_recognition")
+def entity_recognition(req: func.HttpRequest) -> func.HttpResponse:
+    logging.info("calling the entity recognition endpoint")
+    request_json = dict(req.get_json())
+    input_values = []
+    api_key = None
+    try:
+        jsonschema.validate(request_json, schema=get_request_schema())
+        headers_as_dict = dict(req.headers)
+        scenario = headers_as_dict.get("scenario")
+        if scenario != "entity-recognition":
+            raise ValueError(
+                f"incorrect scenario in header. Expected 'entity-recognition', but got {scenario}"
+            )
+        input_values = request_json.get("values")
+        if not input_values:
+            raise ValueError(
+                f"expected values in the request body, but got {input_values}"
+            )
+        api_key = os.getenv("AZURE_INFERENCE_CREDENTIAL")
+        if not api_key:
+            raise ValueError(
+                f"expected an api key from env variable - AZURE_INFERENCE_CREDENTIAL, but got: {api_key}"
+            )
+    except jsonschema.exceptions.ValidationError as e:
+        return func.HttpResponse("Invalid request: {0}".format(e), status_code=400)
+    except ValueError as value_error:
+        return func.HttpResponse(
+            "Invalid request: {0}".format(value_error), status_code=400
+        )
+    response_values = []
+    # TODO: this should be parallelized in the future for performance improvements since we don't need the requests to occur serially
+    for request_body in input_values:
+        api_response = call_chat_completion_model(
+            request_body, api_key
+        )
+        response_values.append(api_response)
+    response_body = {"values": response_values}
+    response = func.HttpResponse(
+        json.dumps(response_body, default=lambda obj: obj.__dict__)
+    )
+    response.headers["Content-Type"] = "application/json"
+    return response
+
+
+# TODO: figure out how to add this into a different file later. It's currently causing interpreter errors when running locally.
+def call_chat_completion_model(request_body: dict, api_key: str):
+    client = ChatCompletionsClient(
+        endpoint='https://azs-grok-phi-3-5-vision.eastus.models.ai.azure.com',

```

Comment:
Make that an easily editable constant, or even better a setting through an env variable, with this value as default?